File: Class_16_Final_and_Reflection/final_problems/linked_list/linkedListStarterCode.js
class Node {
    constructor(value) {
        this.data = value
        this.next = null
    }
}

class LinkedList {
    constructor() {
        this.head = null
    }

    insert(value) {
        const newNode = new Node(value)

        if (this.head) {
            let current = this.head
            
            while(current.next) {
                current = current.next
            }

            current.next = newNode
        } else {
            this.head = newNode
        }
    }
}

let ll = new LinkedList()
ll.insert(1)
ll.insert(2)
ll.insert(3)
ll.insert(4)
ll.insert(5)

File: Unit_15_Final/final_problems/linked_list/linkedListStarterCode.py
class Node:
  def __init__(self, data = None, next=None): 
    self.data = data
    self.next = next

class LinkedList:
  def __init__(self):  
    self.head = None
  
  def insert(self, data):
    newNode = Node(data)
    if(self.head):
      current = self.head
      while(current.next):
        current = current.next
      current.next = newNode
    else:
      self.head = newNode

LL = LinkedList()
LL.insert(1)
LL.insert(2)
LL.insert(3)
LL.insert(4)
LL.insert(5)
